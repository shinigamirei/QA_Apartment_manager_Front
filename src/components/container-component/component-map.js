import { AdminHome, Admin1, Admin2, Admin3, Admin4, Admin5, Admin6,
         RecruiterHome, Recruiter1, Recruiter2, Recruiter3, Recruiter4, Recruiter5, Recruiter6 } from'./placeholder-components';
import Login from './login.component'
import React, { Component }  from 'react';
import ApartmentLanding from '../apartment_manager_landing.component';




let func = function(x){console.log(x)}


export const myConfig =
{
    "null": <Login/>,
    "recruiter": {
        "home": <RecruiterHome/>,
        "buttons": [
                        {'name': 'Home', 'content': <RecruiterHome/>}
                   ],
        "dropdowns":[
                        {   'name': 'Bursary',
                            'content':[
                                        {'name': 'Recruiter Bursary Link 1', 'content': <Recruiter1/>},
                                        {'name': 'Recruiter Bursary Link 2', 'content': <Recruiter2/>}
                                    ]
                        },
                        {  'name': 'Apartments',
                           'content':[
                                        {'name': 'Recruiter Apartment Link 1', 'content': <Recruiter3/>},
                                        {'name': 'Recruiter Apartment Link 2', 'content': <Recruiter4/>}
                                    ]
                        }
                    ],
        "side": [
                        {'name': 'Recruiter SideBar Link 1', 'content': <Recruiter5/>},
                        {'name': 'Recruiter SiderBar Link 2', 'content': <Recruiter6/>}
                   ],
    },
    "admin": {
        "home": <ApartmentLanding className="ApartmentLandingComponent"></ApartmentLanding>,
        "buttons": [
                        {'name': 'Home', 'content': <ApartmentLanding className="ApartmentLandingComponent"></ApartmentLanding>}
                   ],
        "dropdowns":[
                        {   'name': 'Demo Dropdown',
                            'content':[
                                        {'name': 'Admin placeholder 1', 'content':<AdminHome id="helloworld"/>},
                                        {'name': 'Admin placeholder 2', 'content':<Admin1/>},
                                        {'name': 'Admin placeholder 3', 'content': <Admin2/>},
                                        {'name': 'Admin placeholder 4', 'content': <Admin3/>},
                                        {'name': 'Admin placeholder 5', 'content': <Admin4/>}
                                      ]
                        }
                    ],
        "side": [
                        {'name': 'Admin Sidebar Link 1', 'content': <Admin5/>},
                        {'name': 'Admin Sidebar Link 2', 'content': <Admin6/>}
                   ],
    }
}