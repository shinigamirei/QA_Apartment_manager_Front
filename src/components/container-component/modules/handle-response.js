import { authService } from './authService';

export function handleResponse (response) {
	return response.text().then( text => {
	const data = text && JSON.parse(text);
	if (!response.ok){
		if ([401, 403].indexOf(response.status) !== -1){
		authService.logout();
		window.location.hostname.reload(true);
		}
		// auto logout if a 401 or 403 Foribidden response
		const error = (data && data.message) || response.statusText;
		return Promise.reject(error);
		}
		return data; 
	});
}