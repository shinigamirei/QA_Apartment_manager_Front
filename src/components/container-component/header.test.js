import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import renderer from 'react-test-renderer';
import Navigation from './header.component';
import { myConfig } from './component-map';
import { exportAllDeclaration } from '@babel/types';
import { authService } from "./modules/authService";
import sinon from "sinon";

configure({ adapter: new Adapter() });


describe('Header', () => {

    it('should render with "Logout" if user is defined', () => {

        let component = shallow(<Navigation />);

        component.setState({
            currentUser: {
                token: {
                    exp: 1562606357,
                    iat: 1562596277,
                    role: "admin",
                    status: "Active",
                    _id: "5d0a7134d1a499765385ee9f"
                }
            }
        });

        expect(component).toMatchSnapshot();

    });

    it('should render with "Login" if user is null', () => {

        let component = shallow(<Navigation />);

        component.setState({ currentUser: null });

        expect(component).toMatchSnapshot();
    });


    it('should display correct user name', () => {

        let component = shallow(<Navigation />);

        component.setState({
            currentUser: {
                token: {
                    exp: 1562606357,
                    iat: 1562596277,
                    role: "admin",
                    status: "Active",
                    _id: "5d0a7134d1a499765385ee9f"
                }
            }
        });

        component.setState({
            staff_fname: 'Test',
            staff_lname: 'Account'
        });



        expect(component).toMatchSnapshot();
    });



    it('name of user is set on mount', async () => {

        const spy = jest.spyOn(Navigation.prototype, 'componentDidMount');
        const component = mount(<Navigation />);

        component.setState({
            currentUser: {
                token: {
                    exp: 1562606357,
                    iat: 1562596277,
                    role: "admin",
                    status: "Active",
                    _id: "5d0a7134d1a499765385ee9f"
                }
            }
        });
        await component.update();
        component.instance().componentDidMount();
        expect(spy).toHaveBeenCalled();
        // expect(component.state().staff_fname).toEqual("Main")
        component.unmount();
        spy.mockClear();
    });

    it('page redirects for bad credentials', () => {
        const component = mount(<Navigation />);
        component.setState({
            currentUser: {
                token: {
                    exp: 1562606357,
                    iat: 1562596277,
                    role: "admin",
                    status: "Active",
                    _id: "1234"
                }
            }
        });
        
        expect(component).toMatchSnapshot();
        component.unmount();

    });

    it('should component should return null if user is null', () => {


        const spy = jest.spyOn(Navigation.prototype, 'componentDidMount');
        const component = mount(<Navigation />);
        component.setState({
            currentUser: null
        });
        component.instance().componentDidMount();
        expect(component.state('currentUser')).toEqual(null);
        component.unmount();
        spy.mockClear();

    });

    it('should logout', () => {
        const spy = jest.spyOn(Navigation.prototype, 'logout');
        const component = mount(<Navigation />);

        component.setState({
            currentUser: {
                token: {
                    exp: 1562606357,
                    iat: 1562596277,
                    role: "admin",
                    status: "Active",
                    _id: "5d0a7134d1a499765385ee9f"
                }
            }
        });
        component.find('#logoutBtn').first().simulate('click');


        expect(spy).toHaveBeenCalled();

        component.unmount();
    });

    it('should login', () => {
        const spy = jest.spyOn(Navigation.prototype, 'login');
        const component = mount(<Navigation />);

        component.setState({currentUser: null});
        component.find('#loginBtn').first().simulate('click');

        expect(spy).toHaveBeenCalled();

        component.unmount();

    });

});



