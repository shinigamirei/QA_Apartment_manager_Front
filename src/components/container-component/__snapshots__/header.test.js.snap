// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header page redirects for bad credentials 1`] = `
<Navigation>
  <div
    id="bar"
  >
    <div
      id="navigation-bar"
    >
      <Navbar
        color="light"
        expand="md"
        light={true}
        tag="nav"
      >
        <nav
          className="navbar navbar-expand-md navbar-light bg-light"
        >
          <NavbarBrand
            href="/"
            tag="a"
          >
            <a
              className="navbar-brand"
              href="/"
            >
              <img
                alt="QA logo"
                src="QA_logo.png"
                width="60px"
              />
            </a>
          </NavbarBrand>
          <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <ul
              className="ml-auto navbar-nav"
            >
              <NavItem
                className="display_name"
                tag="li"
              >
                <li
                  className="display_name nav-item"
                >
                  Logged in as: 
                   
                   
                   
                    |  
                </li>
              </NavItem>
              <NavItem
                tag="li"
              >
                <li
                  className="nav-item"
                >
                  <Button
                    color="secondary"
                    href="/login"
                    id="logoutBtn"
                    onClick={[Function]}
                    tag="button"
                  >
                    <a
                      aria-label={null}
                      className="btn btn-secondary"
                      href="/login"
                      id="logoutBtn"
                      onClick={[Function]}
                    >
                      Logout
                    </a>
                  </Button>
                </li>
              </NavItem>
            </ul>
          </Nav>
        </nav>
      </Navbar>
    </div>
  </div>
</Navigation>
`;

exports[`Header should display correct user name 1`] = `
<div
  id="bar"
>
  <div
    id="navigation-bar"
  >
    <Navbar
      color="light"
      expand="md"
      light={true}
      tag="nav"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        <img
          alt="QA logo"
          src="QA_logo.png"
          width="60px"
        />
      </NavbarBrand>
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          className="display_name"
          tag="li"
        >
          Logged in as: 
           
           
          Test
           
          Account
            |  
        </NavItem>
        <NavItem
          tag="li"
        >
          <Button
            color="secondary"
            href="/login"
            id="logoutBtn"
            onClick={[Function]}
            tag="button"
          >
            Logout
          </Button>
        </NavItem>
      </Nav>
    </Navbar>
  </div>
</div>
`;

exports[`Header should render with "Login" if user is null 1`] = `
<div
  id="navigation-bar"
>
  <Navbar
    color="light"
    expand="md"
    light={true}
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      <img
        src="QA_logo.png"
        width="60px"
      />
    </NavbarBrand>
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <Button
        color="secondary"
        href="/login"
        id="loginBtn"
        onClick={[Function]}
        tag="button"
      >
        Login
      </Button>
    </Nav>
  </Navbar>
</div>
`;

exports[`Header should render with "Logout" if user is defined 1`] = `
<div
  id="bar"
>
  <div
    id="navigation-bar"
  >
    <Navbar
      color="light"
      expand="md"
      light={true}
      tag="nav"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        <img
          alt="QA logo"
          src="QA_logo.png"
          width="60px"
        />
      </NavbarBrand>
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          className="display_name"
          tag="li"
        >
          Logged in as: 
           
           
           
            |  
        </NavItem>
        <NavItem
          tag="li"
        >
          <Button
            color="secondary"
            href="/login"
            id="logoutBtn"
            onClick={[Function]}
            tag="button"
          >
            Logout
          </Button>
        </NavItem>
      </Nav>
    </Navbar>
  </div>
</div>
`;
